tests/test_factory.py

```diff
@@ -233,3 +233,29 @@ def test_create_poetry_fails_with_invalid_dev_dependencies_iff_with_dev_is_true(
     Factory().create_poetry(
         fixtures_dir / "project_with_invalid_dev_deps", with_groups=False
     )
+
+
+def test_create_poetry_with_groups_and_legacy_dev():
+    poetry = Factory().create_poetry(
+        fixtures_dir / "project_with_groups_and_legacy_dev"
+    )
+
+    package = poetry.package
+    dependencies = package.all_requires
+
+    assert len(dependencies) == 2
+    assert {dependency.name for dependency in dependencies} == {"pytest", "pre-commit"}
+
+
+def test_create_poetry_with_groups_and_explicit_default():
```

Comment:
Fwiw, I think we should add the explicit example. Because that should avoid the issue where folks complain they cannot create a group named "default" or that it does not work as expected. Because as things stand `tool.poetry.dependencies` is a short hand for `tool.poetry.groups.default.dependencies`.

Reply:
@radoering  I am going to resolve this as the suggested change is to be done in poetry docs. Let me know if I misinterpreted.